src/varreplacer.cpp

```diff
@@ -159,6 +159,26 @@ void VarReplacer::update_vardata_and_activities(
     if (orig_act_maple + repl_with_act_maple >= orig_act_maple) {
         solver->var_act_maple[replaced_with] += orig_act_maple;
     }
+
+    assert(orig <= solver->nVars() && replaced_with <= solver->nVars());
+
+    uint32_t neg_orig = Lit(orig,true).toInt();
+    uint32_t pos_orig = Lit(orig,false).toInt();
+    uint32_t neg_replaced_with = Lit(replaced_with,true).toInt();
+    uint32_t pos_replaced_with = Lit(replaced_with,false).toInt();
+
+    double orig_act_lsids_pos = solver->lit_act_lsids[pos_orig];
+    double orig_act_lsids_neg = solver->lit_act_lsids[neg_orig];
+    double repl_with_act_lsids_pos = solver->lit_act_lsids[pos_replaced_with];
+    double repl_with_act_lsids_neg = solver->lit_act_lsids[neg_replaced_with];
+
+    if (orig_act_lsids_pos + repl_with_act_lsids_pos >= orig_act_lsids_pos
+     && orig_act_lsids_neg + repl_with_act_lsids_neg >= orig_act_lsids_neg
+    ) { // WTF is this logic?
```

Comment:
It makes sure when we replace the variable, we accumulate its old VSIDS/Maple scores so we don't loose track of what was the score by replacing. You should do an accumulate of pos&neg here.

So you should not have the "if" and you should just accumulate. But accumulate pos+pos (resp neg+neg) when X is replaced with Y, and accumulate pos+neg(resp. neg+pos) when X is replaced by ~Y.

Reply:
Here variables are replaced with variables. So, "X is replaced by ~Y" - this can't be the case. In fact, we should replace X with Y and ~X with ~Y. Right?